.dotnet/tests/Batch/BatchTests.cs

```diff
@@ -22,32 +23,71 @@ public BatchTests(bool isAsync) : base(isAsync)
     }
 
     [Test]
-    public async Task ListBatchesProtocol()
+    public void ListBatchesProtocol()
     {
         BatchClient client = GetTestClient();
-        ClientResult result = IsAsync
-            ? await client.GetBatchesAsync(after: null, limit: null, options: null)
-            : client.GetBatches(after: null, limit: null, options: null);
+        IEnumerable<ClientResult> pageResults = client.GetBatches(after: null, limit: null, options: null);
 
-        BinaryData response = result.GetRawResponse().Content;
-        JsonDocument jsonDocument = JsonDocument.Parse(response);
-        JsonElement dataElement = jsonDocument.RootElement.GetProperty("data");
+        int count = 0;
+        foreach (ClientResult pageResult in pageResults)
+        {
+            BinaryData response = pageResult.GetRawResponse().Content;
+            JsonDocument jsonDocument = JsonDocument.Parse(response);
```

Comment:
nit: It's good practice to add the `using` keyword before JsonDocument so they get disposed correctly.